src/metabase/models/task_history.clj

```diff
@@ -64,16 +65,41 @@
   {:task_details mi/transform-json-eliding
    :status       mi/transform-keyword})
 
+(defn- filter->where
+  [{:keys [status task] :as filter}]
+  (when task
+    (assert (string? task)))
+  (when status
+    (assert (keyword? status)))
+  (when (not-empty filter)
+    (let [task-filter (when task [:= :task task])
+          status-filter (when status [:= :status (name status)])]
+      {:where (if (and task-filter status-filter)
+                (conj [:and] task-filter status-filter)
+                (or task-filter status-filter))})))
```

Comment:
I'd do something way dumber:
```
(cond-> [:and]
  task (conj [:= :task task])
  status (conj [:= :status status]))
```

a bit more nesting but result is practically the same. Also assertions... idk, malli has checked that already, right?